### PR TITLE
v1.8 backport: node: Remove check whether nextHop is in same L2 

### PR DESCRIPTION
**Pull request by brb**
This is a backport of #14354 for v1.8.

### CODE REVIEW

**Comment by brb** [comment]
test-backport-1.8

**Comment by brb** [comment]
test-backport-1.8

**Review by aanm — APPROVED** [review]

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -554,6 +554,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 		return
 	}
 
+	srcIP := make(net.IP, 4)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]:
```
srcIP := make(net.IP, net.IPv4len)
```

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -575,6 +576,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 			// a gateway. Send arping to the gw IP addr instead of newNode IP addr.
 			// NOTE: we currently don't handle multipath, so only one gw can be used.
 			copy(nextHopIPv4, route.Gw)
+			copy(srcIP, route.Src)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]: We don't know if `route.Src` as size 4. We need to either make `srcIP` with size `net.IPv6Len` or do:
```
routeSrcIPv4 := route.Src.To4()
```

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/datapath/linux/node.go [hunk 3]
```diff
@@ -575,6 +576,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 			// a gateway. Send arping to the gw IP addr instead of newNode IP addr.
 			// NOTE: we currently don't handle multipath, so only one gw can be used.
 			copy(nextHopIPv4, route.Gw)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]: `route.Gw.To4()`?

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by brb** [comment]
test-backport-1.8

**Review by aanm — APPROVED** [review]

**Comment by brb** [comment]
test-backport-1.8

**Comment by brb** [comment]
test-backport-1.8

**Comment by pchaigno** [comment]
Travis CI is currently failing on v1.8; #14531 should fix it. In the meantime, I've manually run the unit tests.

**Review by pchaigno — APPROVED** [review]
As in the v1.9 backport PR, the backported commits are missing references to the upstream commits. You can update and I'll merge without re-running e2e tests, or I'll merge before end of day otherwise.

**Pull request merged**